WordPress/src/main/java/org/wordpress/android/ui/posts/editor/StoriesEventListener.kt

```diff
@@ -61,10 +64,25 @@ class StoriesEventListener @Inject constructor(
     var storiesSavingInProgress = HashSet<StoryIndex>()
         private set
 
-    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
-    private fun onCreate() {
-        dispatcher.register(this)
-        eventBusWrapper.register(this)
+    data class StoryFrameMediaUploadedEvent(
+        val localId: LocalId,
+        val assignedMediaId: RemoteId,
+        val oldUrl: String,
+        val newUrl: String
+    )
+
+    fun startListening() {
+        if (!eventBusWrapper.isRegistered(this)) {
+            dispatcher.register(this)
+            eventBusWrapper.register(this)
+        }
+    }
+
+    fun pauseListening() {
+        if (eventBusWrapper.isRegistered(this)) {
+            dispatcher.unregister(this)
+            eventBusWrapper.unregister(this)
+        }
```

Comment:
@mzorz I am just curious about the use of the `isRegistered` of the `EventBusWrapper` to govern both EventBuses. Overall, I think this implementation is fine especially since the dispatcher is being called before the `eventBusWrapper`. I am just wondering if we would want to expose the `dispatcher`  `isRegistered` method.

Reply:
Nice question there - yes so basically the dispatcher is only used if events from the eventBusWrapper are being listened to so, we could say it's implicit in the logic of this class that actions on FluxC will only be triggered by external events we want the eventBusWrapper to listen to.
Also, we know in this class we register/unregister both to the dispatcher and the eventBusWrapper in tandem so, it is unlikely to have this instance of the class registered to one but not the other.

In order to have 1:1 logic we could do as you suggest but I think it's not needed.

What do you think?

Reply:
Indeed! I think we don't necessarily need the 1:1 logic either. I just wanted to ensure that we wouldn't have any unintended side effects, so it's fine by me.